samples/bluetooth/broadcast_assistant/src/main.c

```diff
@@ -0,0 +1,387 @@
+		if (pos + sub_str_len > str_len) {
+			return false;
+		}
+
+		if (strncasecmp(substr, &str[pos], sub_str_len) == 0) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static void scan_recv_cb(const struct bt_le_scan_recv_info *info,
+			 struct net_buf_simple *ad)
+{
+	int err;
+	struct scan_recv_info sr_info = {0};
+
+	if (scanning_for_broadcast_source) {
+		/* Scan for and select Broadcast Source */
+
+		sr_info.broadcast_id = INVALID_BROADCAST_ID;
+
+		/* We are only interested in non-connectable periodic advertisers */
+		if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) != 0 ||
+		    info->interval == 0) {
+			return;
+		}
+
+		bt_data_parse(ad, device_found, (void *)&sr_info);
+
+		if (sr_info.broadcast_id != INVALID_BROADCAST_ID) {
+			printk("Broadcast Source Found:\n");
+			printk("  BT Name:        %s\n", sr_info.bt_name);
+			printk("  Broadcast Name: %s\n", sr_info.broadcast_name);
+			printk("  Broadcast ID:   0x%06x\n\n", sr_info.broadcast_id);
+
+			if (strlen(CONFIG_SELECT_SOURCE_NAME) > 0U) {
+				/* Compare names with CONFIG_SELECT_SOURCE_NAME */
+				if (is_substring(CONFIG_SELECT_SOURCE_NAME, sr_info.bt_name) ||
+				    is_substring(CONFIG_SELECT_SOURCE_NAME,
+				    sr_info.broadcast_name)) {
+					printk("Match found for '%s'\n", CONFIG_SELECT_SOURCE_NAME);
+				} else {
+					printk("'%s' not found in names\n\n",
+					       CONFIG_SELECT_SOURCE_NAME);
+					return;
+				}
+			}
+
+			err = bt_le_scan_stop();
+			if (err != 0) {
+				printk("bt_le_scan_stop failed with %d\n", err);
+			}
+
+			printk("Selecting Broadcast ID: 0x%06x\n", sr_info.broadcast_id);
+
+			selected_broadcast_id = sr_info.broadcast_id;
+			selected_sid = info->sid;
+			selected_pa_interval = info->interval;
+			bt_addr_le_copy(&selected_addr, info->addr);
+
+			k_sem_give(&sem_source_discovered);
+		}
+	} else {
+		/* Scan for and connect to Broadcast Sink */
+
+		/* We are only interested in connectable advertisers */
+		if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) == 0) {
+			return;
+		}
+
+		bt_data_parse(ad, device_found, (void *)&sr_info);
+
+		if (sr_info.has_bass && sr_info.has_pacs) {
+			printk("Broadcast Sink Found:\n");
+			printk("  BT Name:        %s\n", sr_info.bt_name);
+
+			if (strlen(CONFIG_SELECT_SINK_NAME) > 0U) {
+				/* Compare names with CONFIG_SELECT_SINK_NAME */
+				if (is_substring(CONFIG_SELECT_SINK_NAME, sr_info.bt_name)) {
+					printk("Match found for '%s'\n", CONFIG_SELECT_SINK_NAME);
+				} else {
+					printk("'%s' not found in names\n\n",
+						   CONFIG_SELECT_SINK_NAME);
+					return;
+				}
+			}
+
+			err = bt_le_scan_stop();
+			if (err != 0) {
+				printk("bt_le_scan_stop failed with %d\n", err);
+			}
+
+			printk("Connecting to Broadcast Sink: %s\n", sr_info.bt_name);
+
+			err = bt_conn_le_create(info->addr, BT_CONN_LE_CREATE_CONN,
+				BT_LE_CONN_PARAM_DEFAULT,
+				&broadcast_sink_conn);
+			if (err != 0) {
+				printk("Create conn to failed (%u)\n", err);
```

Comment:
Did you mean to add the address after the `to` here?

Reply:
maybe some strange copy&paste... fixed